src/librados/RadosClient.cc

```diff
@@ -94,30 +94,68 @@ int64_t librados::RadosClient::lookup_pool(const char *name)
 
 bool librados::RadosClient::pool_requires_alignment(int64_t pool_id)
 {
+  bool requires;
+  int r = pool_requires_alignment2(pool_id, &requires);
+  if (r < 0) {
+    return r;
+  }
+
+  return requires;
+}
+
+// a safer version of pool_requires_alignment
+int librados::RadosClient::pool_requires_alignment2(int64_t pool_id,
+	bool *requires)
+{
+  if (!requires)
+    return -EINVAL;
+
   int r = wait_for_osdmap();
   if (r < 0) {
     return r;
   }
 
   const OSDMap *osdmap = objecter->get_osdmap_read();
-  bool ret = osdmap->have_pg_pool(pool_id) &&
-    osdmap->get_pg_pool(pool_id)->requires_aligned_append();
+  if (!osdmap->have_pg_pool(pool_id)) { 
+    objecter->put_osdmap_read();
+    return -ENOENT;
+  }
+  *requires = osdmap->get_pg_pool(pool_id)->requires_aligned_append();
   objecter->put_osdmap_read();
-  return ret;
+  return 0;
 }
 
 uint64_t librados::RadosClient::pool_required_alignment(int64_t pool_id)
 {
+  uint64_t alignment;
+  int r = pool_required_alignment2(pool_id, &alignment);
+  if (r < 0) {
+    return r;
```

Comment:
One problem here.  Return a negative int as uint64_t.  What does caller do with a huge number?  Should it return 0 and caller should check for that as an error indication?